pkg/management/catalog/catalog.go

```diff
@@ -94,62 +91,62 @@ func (catalog *Catalog) FindClosestBackupInfo(recoveryTarget *v1.RecoveryTarget)
 	return catalog.findlatestBackupFromTimeline(targetTLI), nil
 }
 
-func (catalog *Catalog) findClosestBackupFromTargetLSN(
+func (catalog Catalog) findClosestBackupFromTargetLSN(
 	targetLSNString string,
 	targetTLI string,
 ) (*BarmanBackup, error) {
 	targetLSN := postgres.LSN(targetLSNString)
 	if _, err := targetLSN.Parse(); err != nil {
 		return nil, fmt.Errorf("while parsing recovery target targetLSN: " + err.Error())
 	}
-	for i := len(catalog.List) - 1; i >= 0; i-- {
-		barmanBackup := catalog.List[i]
+	for i := len(catalog) - 1; i >= 0; i-- {
+		barmanBackup := catalog[i]
 		if !barmanBackup.isBackupDone() {
 			continue
 		}
 		if (strconv.Itoa(barmanBackup.TimeLine) == targetTLI ||
 			// if targetTLI is not an integer, it will be ignored actually
 			currentTLIRegex.MatchString(targetTLI)) &&
 			postgres.LSN(barmanBackup.BeginLSN).Less(targetLSN) {
-			return &catalog.List[i], nil
+			return &catalog[i], nil
 		}
 	}
 	return nil, nil
 }
 
-func (catalog *Catalog) findClosestBackupFromTargetTime(
+func (catalog Catalog) findClosestBackupFromTargetTime(
 	targetTimeString string,
 	targetTLI string,
 ) (*BarmanBackup, error) {
 	targetTime, err := utils.ParseTargetTime(nil, targetTimeString)
 	if err != nil {
 		return nil, fmt.Errorf("while parsing recovery target targetTime: " + err.Error())
 	}
-	for i := len(catalog.List) - 1; i >= 0; i-- {
-		barmanBackup := catalog.List[i]
+	for i := len(catalog) - 1; i >= 0; i-- {
+		barmanBackup := catalog[i]
 		if !barmanBackup.isBackupDone() {
 			continue
 		}
 		if (strconv.Itoa(barmanBackup.TimeLine) == targetTLI ||
 			// if targetTLI is not an integer, it will be ignored actually
 			currentTLIRegex.MatchString(targetTLI)) &&
-			barmanBackup.EndTime.Before(targetTime) {
-			return &catalog.List[i], nil
+			barmanBackup.BeginTime.Before(targetTime) {
+			return &catalog[i], nil
```

Comment:
this is reverting https://github.com/cloudnative-pg/cloudnative-pg/commit/216cc54a5a215b70d63725ea1d1ef637410ff223, is this intentional?

Reply:
eeeek. No! Rebase artifact.. Thanks for catching @phisco